packages/suite/src/hooks/wallet/trading/form/useTradingExchangeForm.ts

```diff
@@ -578,6 +586,76 @@ export const useTradingExchangeForm = ({
         }
     };
 
+    const signDataAndConfirm = async () => {
+        if (!selectedQuote?.signData) {
+            dispatch(
+                notificationsActions.addToast({
+                    type: 'error',
+                    error: 'Cannot sign, missing data',
+                }),
+            );
+
+            return;
+        }
+
+        dispatch(tradingCommonActions.setTradingModalAccountKey(account.key));
+
+        let signature;
+        if (
+            account.networkType === 'ethereum' &&
+            selectedQuote.signData.type === 'eip712-typed-data'
+        ) {
+            const typedData = selectedQuote?.signData
+                .data as EthereumSignTypedDataMessage<EthereumSignTypedDataTypes>;
+
+            const result = await TrezorConnect.ethereumSignTypedData({
+                device,
+                path: account.path,
+                metamask_v4_compat: false,
+                data: typedData,
+            });
+
+            if (!result.success) {
+                dispatch(
+                    notificationsActions.addToast({
+                        type: 'sign-message-error',
+                        error: result.payload.error,
+                    }),
+                );
+
+                return;
+            }
+
+            signature = result.payload.signature;
+        } else {
+            dispatch(
+                notificationsActions.addToast({
+                    type: 'error',
+                    error: 'Cannot sign data, unsupported network',
+                }),
+            );
+
+            return;
+        }
+
+        const trade = { ...selectedQuote };
+        trade.signature = signature;
+        trade.status = 'SIGN_DATA';
+
+        if (!trade.receiveAddress) {
+            return;
+        }
+
+        dispatch(
+            tradingExchangeActions.saveTrade(
+                trade,
+                selectedAccount.account,
+                new Date().toISOString(),
+            ),
+        );
+        confirmTrade(trade.receiveAddress, undefined, trade);
```

Comment:
Let's await the Promise

Reply:
fixed in https://github.com/trezor/trezor-suite/pull/17005/commits/18d74b1db384e3076aeaeae1518ceeab53e5f89a